capstone-frontend/src/components/BookSearchList.jsx

```diff
@@ -19,29 +18,22 @@ class BookSearchList extends Component {
       .catch(err => console.log(err));
   }
 
-  getUserBookLists = async () => {
-
-    const userID = window.localStorage.getItem("userID");
-
-    const bookLists = await fetch(`/api/booklist?userID=${userID}`, {
-      method: "GET",
-    }).then(resp => resp.json());
-
-    this.setState({ bookLists });
-  }
-
-
   componentDidMount() {
     this.getData();
-    this.getUserBookLists();
+  }
+
+  componentWillUnmount() {
+    this.setState = (state,callback) => {
```

Comment:
What does this do?

Reply:
I think it remounts the Component.

Reply:
It makes it so that the `setState` method of React's base `Component` class won't be called, as this will be called instead. You should be able to remove this though as long as you change the `component` to a `render` prop in your routes above (see other comment).